src/main/java/catan/services/util/play/EndTurnUtil.java

```diff
@@ -1,12 +1,44 @@
 package catan.services.util.play;
 
 import catan.domain.model.game.GameBean;
+import catan.domain.model.game.types.GameStage;
 import org.springframework.stereotype.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
 
 @Component
 public class EndTurnUtil {
+    private Logger log = LoggerFactory.getLogger(EndTurnUtil.class);
+
+    public Integer endTurnImplInPreparationStage(GameBean game, List<List<String>> initialBuildingsSet) {
+        Integer nextMoveNumber;
+        if (isCycleFinished(game)) {
+            if (isLastCycle(game, initialBuildingsSet.size())) {
+                game.setStage(GameStage.MAIN);
+                game.setPreparationCycle(null);
+                game.setCurrentCycleBuildingNumber(0);
+                log.debug("Game Stage was changed from PREPARATION to {}", game.getStage());
+                nextMoveNumber = 1;
+            } else {
+                nextMoveNumber = getNextMoveInPreparationStage(game);
```

Comment:
you place setting of nextMoveNumber in beggining of 'else' condition here and at the end of 'if' condition above, we should have the same place for it